sretoolbox/binaries/mtcli.py

```diff
@@ -30,10 +31,13 @@ class Mtcli(Binary):
     """
 
     binary_template = "mtcli-{version}"
+    system = platform.system()
+    machine = platform.machine()
     download_url_template = (
         "https://github.com/mt-sre/addon-metadata-operator/"
         "releases/download/v{major}.{minor}.{patch}/"
-        "mtcli_{major}.{minor}.{patch}_Linux_x86_64.tar.gz"
+        "mtcli_{major}.{minor}.{patch}_"
+        f"{system}_{machine}.tar.gz"
```

Comment:
This seems to work, from my local testing `platform.system() == "Linux"` and `platform.machine() == "x86_64"`. We need to coordinate whatever these values return with the goreleaser replacements: https://github.com/mt-sre/addon-metadata-operator/blob/master/.goreleaser.yml#L28

Also, as a follow up on this, I created a ticket to investigate modifying the `addon-metadata-operator` code so we can drop `go-sqlite3`, because this is the only library forcing us to do a C build instead of pure Go code build, which prevents us from cross-compiling: https://issues.redhat.com/browse/MTSRE-329

Reply:
> We need to coordinate whatever these values return with the goreleaser replacements: https://github.com/mt-sre/addon-metadata-operator/blob/master/.goreleaser.yml#L28

Already did for Linux and Darwin (both x86_64 (amd64) and arm64) as well, works well :) Though, had a hard time confirming for i386 (ssh-ing into i386 VMs seems to be a mess) although unknowingly I ended doing for i686 (`platform.machine() -> 'i686') but I feel we don't need worry much about i386 considering how old generation it corresponds to and highly unlikely it will be used.

wdyt :)

Reply:
> Also, as a follow up on this, I created a ticket to investigate modifying the addon-metadata-operator code so we can drop go-sqlite3, because this is the only library forcing us to do a C build instead of pure Go code build, which prevents us from cross-compiling: https://issues.redhat.com/browse/MTSRE-329

Perfect!